stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/gallery/internal/AttachmentGalleryVideoPageFragment.kt

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.ui.gallery.internal
+
+import android.content.Context
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.MediaController
+import android.widget.Toast
+import androidx.core.view.updateLayoutParams
+import androidx.fragment.app.Fragment
+import com.getstream.sdk.chat.images.load
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.ui.ChatUI
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamUiItemAttachmentGalleryVideoBinding
+import io.getstream.chat.android.ui.gallery.AttachmentGalleryViewMediaStyle
+import io.getstream.chat.android.ui.gallery.options.AttachmentGalleryOptionsViewStyle
+
+internal class AttachmentGalleryVideoPageFragment : Fragment() {
+
+    private var _binding: StreamUiItemAttachmentGalleryVideoBinding? = null
+    private val binding get() = _binding!!
+
+    /**
+     * Holds the style necessary to stylize the play button.
+     *
+     * Fetching the style depends on [Context] so use this property
+     * only after it has been obtained during or after [onAttach].
+     */
+    private val style by lazy {
+        AttachmentGalleryViewMediaStyle(
+            context = requireContext(),
+            attrs = null
+        )
+    }
+
+    /**
+     * If the video has been prepared and the video player
+     * is ready for playback.
+     *
+     * VideoView does not expose state so we use these to track
+     * state externally.
+     */
+    private var playbackPrepared = false
+        set(value) {
```

Comment:
`VideoView` does not expose it's internal state properly so as a consequence we have these overly complex external state trackers.

I don't like them and I don't think they are bulletproof. I'll add a discussion card to see if we might want to revisit our decision to not depend on media players such as ExoPlayer.

The [info listener](https://developer.android.com/reference/android/media/MediaPlayer.OnInfoListener) unfortunately did not yield any concrete results. It would fire the first time that the video would start, but then ignore any subsequent playback actions. Any sort of state tracking with this View seems to have been an after thought.